src/pkg/packager/creator/normal.go

```diff
@@ -174,37 +178,32 @@ func (pc *PackageCreator) Assemble(dst *layout.PackagePaths, components []types.
 
 		dst.AddImages()
 
-		var pulled []images.ImgInfo
-		var err error
-
 		ctx, cancel := context.WithCancel(context.TODO())
-		doPull := func() error {
-			imgConfig := images.ImageConfig{
-				ImagesPath:        dst.Images.Base,
-				ImageList:         imageList,
-				Insecure:          config.CommonOptions.Insecure,
-				Architectures:     []string{arch},
-				RegistryOverrides: pc.createOpts.RegistryOverrides,
-			}
 
-			pulled, err = imgConfig.PullAll()
-			if errors.Is(err, lang.ErrUnsupportedImageType) {
-				cancel()
-			}
-
-			return err
+		pullCfg := images.PullConfig{
+			DestinationDirectory: dst.Images.Base,
+			References:           imageList,
+			Arch:                 arch,
+			RegistryOverrides:    pc.createOpts.RegistryOverrides,
+			CacheDirectory:       filepath.Join(config.GetAbsCachePath(), layout.ImagesDir),
 		}
 
-		if err := helpers.RetryWithContext(ctx, doPull, 3, 5*time.Second, message.Warnf); err != nil {
-			return fmt.Errorf("unable to pull images: %w", err)
+		pulled, err := images.Pull(ctx, cancel, pullCfg)
+		if err != nil {
+			return err
 		}
 
-		for _, imgInfo := range pulled {
-			if err := dst.Images.AddV1Image(imgInfo.Img); err != nil {
+		for info, img := range pulled {
+			info, img := info, img
```

Comment:
Re-declaring loop variables (in Go 1.21) is only required if you intend to keep reference to them past the loop iteration. It only becomes an issue during concurrent processing, like running go routines in a for loop.

FYI this has been fixed in Go 1.22.

Reply:
linting was being mean to me, so I re-declared, I'll check if I can remove / validate my linting rules